### DISCUSSION TITLE
How to get the source path from a page?

### DISCUSSION THREAD

**Discussion by valtlai** [lumeland/lume#96] [Q&A] [answered]
I would like to implement a *View source* link to view the source of the current page in site’s GitHub repo. I would to something like this:

```njk
<a href="{{ site.repoUrl }}{{ SOMETHING_HERE }}">View source</a>
```

which should produce something like:

```html
<a href="https://github.com/valtlai/valtlai.fi/blob/main/blog/hello.md">View source</a>
```

Is there a good way?

**Comment by valtlai** [comment, marked as answer]
Found a solution. Is this the best approach?

In `_data/site.yml`, I have:
```yml
repoUrl: https://github.com/valtlai/valtlai.fi/blob/main
```

In `_config.js`:
```js
site.preprocess([".html"], (page) => {
  page.data.srcUrl = page.src.path + page.src.ext;
});
```

In a layout file:
```njk
<a href="{{ site.repoUrl }}{{ srcUrl }}">View source</a>
```

**Reply by oscarotero** [reply]
Yes, it's a good approach. Templates don't have access to the page instance (with the properties `src`, `dest`, `data`, `content`), but only to `data` object, so a preprocessor to create a variable with this info is a nice idea.
